### PR TITLE
test: run integrations tests in both OS and ENT where possible

### PR DESCRIPTION
**Pull request by tranchitella**
Rework tests in the following files to run both in open-source and
enterprise:

- test_basic_integration.py
- test_db_migration.py
- test_fault_tolerance.py
- test_security.py
- test_signed_image_update.py
- test_update_modules.py

Changelog: none
Ticket: QA-332

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

Based on: https://github.com/mendersoftware/integration/pull/1920

### CODE REVIEW

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

File: tests/common_setup.py [hunk 1]
```diff
@@ -235,6 +248,117 @@ def enterprise_no_client(request):
...
+    env = container_factory.getEnterpriseSetup(num_clients=0)
+    request.addfinalizer(env.teardown)
+
+    env.setup()
+    reset_mender_api(env)
+
+    tenant = create_tenant(env)
+    mender_device = new_tenant_client(env, "mender-client", tenant["tenant_token"])
+    mender_device.ssh_is_opened()
+    env.device = mender_device
```
Comment (lluiscampos) [inline tests/common_setup.py:262]: `new_tenant_client` already adds the device to the environment, so these lines (and similarly below) could be:
```suggestion
    new_tenant_client(env, "mender-client", tenant["tenant_token"])
    env.device_group.ssh_is_opened()
```

**Review by lluiscampos — APPROVED** [review]
Jaw-dropping work :exploding_head:

**Comment by lluiscampos** [comment]
`test:integration:enterprise` goes from 54 tests (today's nightly run) to 98 :rocket:

File: tests/common_setup.py (hunk 1 above)
Comment (tranchitella) [reply tests/common_setup.py:262]: I will address it in the other PR (https://github.com/mendersoftware/integration/pull/1923/commits/f20c0e80846fb2a1600903e2f28b726b909dcbf3)

**Pull request merged**